lib/tasks/reset_qa.rake

```diff
@@ -0,0 +1,37 @@
+desc 'Reset the database to a state with some standard providers and test applications. Preserves existing support users'
+task reset_qa: :environment do
+  raise 'You can’t do this anywhere but QA or development!' unless HostingEnvironment.qa? || Rails.env.development?
+
+  Rake::Task['_reset_qa'].invoke
+end
+
+task _reset_qa: %i[
+  environment
+  backup_support_users
+  truncate_qa_database
+  restore_support_users
+  sync_dev_providers_and_open_courses
+  generate_test_applications
+]
+
+task truncate_qa_database: :environment do
+  raise 'You can’t do this anywhere but QA or development!' unless HostingEnvironment.qa? || Rails.env.development?
+
+  ActiveRecord::Tasks::DatabaseTasks.truncate_all
```

Comment:
Does this truncate the 'system' tables like `schema_migrations`?

Reply:
Good question. Seems not to. Ran it locally and

![Screenshot 2020-03-04 at 11 36 24](https://user-images.githubusercontent.com/642279/75875851-66784d00-5e0c-11ea-8992-f7c3b5e86a83.png)